### PR TITLE
[system/cpu][system/core] - New config for using performance counters

### PR DESCRIPTION
**Pull request by VihasMakwana**
This PR uses follows up on https://github.com/elastic/elastic-agent-system-metrics/pull/192. 
It makes the use of performance counters to collect cpu metrics on windows.

Docs:
- Added

Test results:
- Shared on https://github.com/elastic/beats/issues/40926

Closes https://github.com/elastic/beats/issues/40926

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -183,7 +183,7 @@ require (
 	github.com/elastic/ebpfevents v0.6.0
 	github.com/elastic/elastic-agent-autodiscover v0.9.0
 	github.com/elastic/elastic-agent-libs v0.17.4
-	github.com/elastic/elastic-agent-system-metrics v0.11.4
+	github.com/elastic/elastic-agent-system-metrics v0.11.5-0.20241210092419-b0ba11859dc9
```
Comment (VihasMakwana) [inline go.mod:186]: This is temporary. Once I verify the fix one more time, I will release a patch and update go.mod

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -127,7 +127,7 @@ filebeat.inputs:
 - Add AWS Cloudwatch capability to retrieve tags from AWS/ApiGateway resources. {pull}40755[40755]
 - Add new metrics for the vSphere Virtualmachine metricset. {pull}40485[40485]
 - Add `metrics_count` to Prometheus module if `metrics_count: true` is set. {pull}40411[40411]
-
+- Add `use_performance_counters` to use performance counters to collect CPU related metrics on Windows. {pull}41965[41965]
```
Comment (belimawr) [inline CHANGELOG.asciidoc:130]: Don't edit the `CHANGELOG.asciidoc`, instead edit the `CHANGELOG.next.asciidoc`.

`CHANGELOG.asciidoc` gets updated during the release process.

Comment (VihasMakwana) [reply CHANGELOG.asciidoc:130]: Oh, sorry. I'll update `CHANGELOG.asciidoc`

**Review by leehinman — APPROVED** [review]

File: CHANGELOG.next.asciidoc [hunk 3]
```diff
@@ -420,6 +420,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Add support for region/zone for Vertex AI service in GCP module {pull}41551[41551]
 - Add support for location label as an optional configuration parameter in GCP metrics metricset. {issue}41550[41550] {pull}41626[41626]
 - Added `tier_preference`, `creation_date` and `version` fields to the `elasticsearch.index` metricset. {pull}41944[41944]
+- Add `use_performance_counters` to use performance counters to collect CPU related metrics on Windows. {pull}41965[41965]
```
Comment (cmacknz) [inline CHANGELOG.next.asciidoc:423]: You should specify which module (system) and metricsets this applies to (core, cpu)

File: metricbeat/docs/modules/system.asciidoc [hunk 4]
```diff
@@ -265,6 +265,11 @@ metricbeat.modules:
 
   # Filter systemd services based on a name pattern
   #service.pattern_filter: ["ssh*", "nfs*"]
+
+  # This option enables the use of performance counters to collect data for cpu/core metricset.
+  # Only effective for Windows.
+  # You should use this option if running beats on machins with more than 64 cores.
+  #use_performance_counters: false
```
Comment (cmacknz) [inline metricbeat/docs/modules/system.asciidoc:272]: We expect to change to default to true once we are confident there are no regressions, correct?

Is it worth just defaulting it to true now, and only turning it off if there is a problem?

This won't be released until 9.0 and 8.18.0 given the current timing.

**Review by cmacknz — APPROVED** [review]
LGTM, though the changelog should be a bit more specific.

I am also fine if we default performance counters to true by default. The important thing is we have a way to turn it back off easily without needing to wait for a new build a Beats in places where it could cause problems (e.g. the integrations repository tests, the elastic agent tests, Elastic cloud monitoring in QA and staging, etc).

**Pull request merged**